apps/ui/pages/applications/[id]/page3.tsx

```diff
@@ -58,6 +58,8 @@ function Page3Form(): JSX.Element | null {
     if (hasRegistration) {
       register("organisation.identifier", { required: true });
     } else {
+      // Unregister does not remove the form value (neither from DOM nor from the form state)
+      setValue("organisation.identifier", undefined);
       unregister("organisation.identifier");
     }
   }, [type, register, unregister]);
```

Comment:
Lint warning:
`React Hook useEffect has a missing dependency: 'setValue'. Either include it or remove the dependency array`